src/ast/ast_validations.c

```diff
@@ -57,6 +58,18 @@ static void _AST_GetIdentifiers(const cypher_astnode_t *node, rax *identifiers)
 	cypher_astnode_type_t type = cypher_astnode_type(node);
 	if(type == CYPHER_AST_PROJECTION) child_count = 1;
 
+	/* In case current node is of type CALL
+	 * process procedure call arguments only
+	 * outputs are not refering to previous identifiers */
+	if(type == CYPHER_AST_CALL) {
+		uint arg_count = cypher_ast_call_narguments(node);
+		for(uint i = 0; i < arg_count; i++) {
+			const cypher_astnode_t *arg = cypher_ast_call_get_argument(node, i);
+			_AST_GetIdentifiers(arg, identifiers);
+		}
+		return;
+	}
+
```

Comment:
move this block to after the type extraction
also, think about adding or rephrasing `process procedure call arguments only` to `process procedure call arguments. They should be defined prior to the procedure call`